V/VMEC/build_tarballs.jl

```diff
@@ -1,43 +1,106 @@
+    ln -s ../../artifacts/${artifact_id}/lib/libmbedcrypto${mbed_version}.${dlext} libmbedcrypto.3.${dlext}
+    ln -s ../../artifacts/${artifact_id}/lib/libmbedcrypto${mbed_version}.${dlext} libmbedcrypto.0.${dlext}
+    ln -s ../../artifacts/${artifact_id}/lib/libmbedx509${mbed_version}.${dlext} libmbedx509.0.${dlext}
+    cd ${WORKSPACE}/srcdir/VMEC*
+  fi
+  # Configure with MKL provided ScaLAPACK
+  ./autogen.sh
+  ./configure CC=mpicc FC=mpifort F77=mpifort FFLAGS=-O3 FCFLAGS=-O3 --with-mkl --enable-tbb --host=$target --target=$target --prefix=$prefix 
+  make && make install && make clean
+  
+  ./configure CC=mpicc FC=mpifort F77=mpifort FFLAGS=-O3 FCFLAGS=-O3 --with-mkl --host=$target --target=$target --prefix=$prefix 
+  make && make install && make clean
+
+  # Configure with OpenBLAS and ScaLAPACK from SCALAPACK_jll
+  ./configure CC=mpicc FC=mpifort F77=mpifort FFLAGS=-O3 FCFLAGS=-O3  --host=$target --target=$target --prefix=$prefix
+  make && make install && make clean
+fi
 """
 
-platforms = [
-    Platform("x86_64", "linux"; libc = "glibc", libgfortran_version="4.0.0"),
-    Platform("x86_64", "linux"; libc = "glibc", libgfortran_version="5.0.0"),
-]
-#platforms = expand_gfortran_versions(platforms)
+incompatible_arch = ["i686", "armv6l", "armv7l", "aarch64", "powerpc64le"]
+incompatible_os = ["freebsd"]
+
+platforms = expand_gfortran_versions(supported_platforms())
+
+# Filter out libgfortran_version = 3.0.0 which is incompatible with VMEC
+platforms = filter(p -> p["libgfortran_version"] != "3.0.0", platforms)
+
+# Filter incompatible architectures and operating systems
+for arch in incompatible_arch
+  global platforms = filter(p -> p.tags["arch"] != arch, platforms)
+end
+
+for os in incompatible_os
+  global platforms = filter(p -> p.tags["os"] != os, platforms)
+end
+
+# Right now VMEC only works with libc=glibc, filter out any musl dependencies
+platforms = filter(p -> (haskey(p.tags, "libc") && p.tags["libc"] != "musl") || !haskey(p.tags, "libc"), platforms)
 
 # The products that we will ensure are always built
+# Don't automatically dl_open so that the appropriate 
+# library can be loaded on intiation of VMEC.jl
 products = [
-    LibraryProduct("libvmec_mkl", :libvmec_mkl, dont_dlopen = true),
-    LibraryProduct("libvmec_openblas", :libvmec_openblas, dont_dlopen = true),
+    LibraryProduct("libvmec_mkl_tbb", :libvmec_mkl_tbb, dont_dlopen = true),
+    LibraryProduct("libvmec_mkl_intel", :libvmec_mkl_intel, dont_dlopen = true),
+    LibraryProduct("libvmec_openblas_pthread", :libvmec_openblas_pthread, dont_dlopen = true),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency("MPICH_jll"),
-    Dependency("OpenBLAS_jll", v"0.3.17"),
+    BuildDependency("MPICH_jll"),
+    BuildDependency("OpenBLAS_jll"),
+    BuildDependency("MKL_jll"),
+    BuildDependency("oneTBB_jll"),
+    BuildDependency("NetCDF_jll"),
+    BuildDependency("NetCDFF_jll"),
```

Comment:
I'm confused, why these are build-only dependencies?

Reply:
I think I misunderstood the difference between `BuildDependency` and `Dependency`.  I was thinking that if it was a `BuildDependency` only, then the host system would provide the necessary libraries and not an artifact.  I'm in the process of changing this back.